timesketch/lib/analyzers/crash.py

```diff
@@ -0,0 +1,178 @@
+            if '.exe' in text.lower():
+                match = regex.search(message)
+                if match:
+                    # The regex can match on full file paths and filenames,
+                    # so only return the filename.
+                    return min([m for m in match.groups() if m])
+            return None
+
+        # For adding information to identified entries.
+        def mark_as_crash(event, filename):
+            event.add_star()
+            event.add_attributes({'crash_app': filename})
+            event.add_label('crash')
+
+        query = formulate_query(query_elements)
+
+        return_fields = ['data_type', 'message', 'filename']
+
+        # Generator of events based on your query.
+        events = self.event_stream(
+            query_string=query, return_fields=return_fields)
+
+        # Get the executable filenames.
+        # Examples: 
+        #   "[...]\WER\ReportQueue\AppCrash_notepad.exe_8d163e29d3960561ca2e9723640cd8fff5c2ad5_cab_0a5810ac"
+        #   "[...]/WER/ReportQueue/NonCritical_x64_d473a376adfb18a7b165c5e3c26de43cd8bccb_cab_07fd2620"
+        #   "Strings: ['iexplore.exe', '8.0.7601.17514' [...]"
+        re_filename = re.compile(r'(?:\\|\/)(?:AppCrash|Critical|NonCritical)_(.+?\.exe)_[a-f0-9]{16,}|\'([^\']+\.exe)\'', re.I)
+
+        # Container for filenames of crashed applications.
+        filenames = dict()
+
+        for event in events:
+
+            data_type = event.source.get('data_type')
+
+            # Search event log entries for filenames of crashed applications.
+            if data_type == 'windows:evtx:record':
+                message = event.source.get('message')
+
+                fn = extract_filename(message, re_filename)
+                if not fn:
+                    continue
+                filenames[fn] = ''
+                mark_as_crash(event, fn)
+
+            # Search file system entries for filenames of crashed applications.
+            elif data_type == 'fs:stat':
+                filename = event.source.get('filename')
+
+                fn = extract_filename(filename, re_filename)
+                if not fn:
+                    continue
+                filenames[fn] = ''
+                mark_as_crash(event, fn)
+
+            # Check if the crash reporting has been disabled in the registry.
+            elif data_type == 'windows:registry:key_value':
+                event.add_star()
+                event.add_comment('WARNING: The crash reporting was disabled. '
+                'It could be indicative of attacker activity. '
+                'For details refer to page 16 from '
+                'https://assets.documentcloud.org/documents/3461560/Google-Aquarium-Clean.pdf.')
+
+            # Commit the event to the datastore.
+            event.commit()
+
+        # Create a saved view with our query.
+        if filenames:
+            self.sketch.add_view('Crash activity', 'app_crashes', query_string=query)
```

Comment:
so if you have he query string `query` then you'll get a whole bunch of events from this view that are not really crashes, since you have checks here, and only if the checks suceced then you add something to the event, for instance of you had added a tag instead of a label you could do  

```
query_string='tag:"crash"`
```

and then you just get the correct events